File: php/features.php
<?php
declare(strict_types=1);

// Usaspending SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UsaspendingFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UsaspendingBaseFeature();
            case "test":
                return new UsaspendingTestFeature();
            default:
                return new UsaspendingBaseFeature();
        }
    }
}

File: py/features.py
# Usaspending SDK feature factory

from feature.base_feature import UsaspendingBaseFeature
from feature.test_feature import UsaspendingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UsaspendingBaseFeature(),
        "test": lambda: UsaspendingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
